cpu/stm32f1/include/periph_cpu.h

```diff
@@ -19,13 +19,88 @@
 #ifndef PERIPH_CPU_H_
 #define PERIPH_CPU_H_
 
+#include "cpu.h"
 #include "periph/dev_enums.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-/* nothing defined here so far... */
+/**
+ * @brief   Overwrite the default gpio_t type definition
+ * @{
+ */
+#define HAVE_GPIO_T
+typedef uint32_t gpio_t;
+/** @} */
+
+/**
+ * @brief   Definition of a fitting UNDEF value
+ */
+#define GPIO_UNDEF          (0xffffffff)
+
+/**
+ * @brief   Define a CPU specific GPIO pin generator macro
+ */
+#define GPIO(x, y)          ((GPIOA_BASE + (x << 10)) | y)
+
+/**
+ * @brief   Override values for pull register configuration
+ * @{
+ */
+#define HAVE_GPIO_PP_T
+typedef enum {
+    GPIO_NOPULL = 4,        /**< do not use internal pull resistors */
+    GPIO_PULLUP = 9,        /**< enable internal pull-up resistor */
+    GPIO_PULLDOWN = 8       /**< enable internal pull-down resistor */
```

Comment:
Could you shortly explain these values? Maybe referring to the [reference manual](http://www.st.com/st-web-ui/static/active/en/resource/technical/document/reference_manual/CD00171190.pdf) (sec.9.2.1, p. 171 or table 20, p.161)? I think it's much faster than figuring this out by myself